[WIP] [EEN-19576] I/O API documentation

skazanyNaGlany opened the pull request:
JIRA:
https://eagleeyenetworks.atlassian.net/browse/EEN-19576

mcotton reviewed (changes requested):
- Mentions of ETags do not belong in the public documentation
- The "Reading I/O events" example doesn't include an actual example
- What event type should they subscribe to?
- Is that event type added in the "Event Objects" section?
- Using the parameter "username" is confusing
- When setting the state of the output port do you send another "camera_settings_add" or "camera_settings_delete"?

skazanyNaGlany commented:
> * Mentions of ETags do not belong in the public documentation
Removed, using Event instead.

> * The "Reading I/O events" example doesn't include an actual example
Added request and response examples.

> * What event type should they subscribe to?
Depends on needs, but most important events are A "active" and H "hold" events, so for example `IIA0` and `IIH0`. Each camera has its own number of I/O ports, so it must be read from `camera_parameters.active_settings.camera_io.d` first.

> * Is that event type added in the "Event Objects" section?
Added, but I'm not sure about it. There is no static I/O event since each camera has its own count of the ports.

> * Using the parameter "username" is confusing
Added more information about this field.

> * When setting the state of the output port do you send another "camera_settings_add" or "camera_settings_delete"?
`camera_settings_add` only.

Fixed the issues, please take a look, thanks.

mcotton commented:
This needs to go through an API review.  Even with your changes it is still confusing and is not consistent with our existing API.

skazanyNaGlany commented:
> This needs to go through an API review. Even with your changes it is still confusing and is not consistent with our existing API.

I'm sorry, but can you explain more? I thought that I wrote it very well. Can we schedule a meeting? I will write you a message.
